### PR TITLE
chore: move different chain hardfork sets to `reth-ethereum-forks`

### PR DESCRIPTION
**Pull request by joshieDo**
Makes them `const` as well.

### CODE REVIEW

**Review by Rjected — COMMENTED** [review]
thinking we might want to rename the crate if this contains op forks as well? maybe reth-hardforks or something?

**Comment by mattsse** [comment]
yeah but we can this next

File: crates/ethereum-forks/src/chains/ethereum.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+use crate::{ForkCondition, Hardfork};
+use alloy_primitives::U256;
+
+const PARIS_TTD_MAINNET: U256 = U256::from_limbs([15566869308787654656, 3184, 0, 0]);
+pub(crate) const PARIS_TTD_GOERLI: U256 = U256::from_limbs([10790000, 0, 0, 0]);
```
Comment (mattsse) [inline crates/ethereum-forks/src/chains/ethereum.rs:5]: why does this have different pub

File: crates/ethereum-forks/src/chains/mod.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+pub mod ethereum;
+
+/// Optimism chains
+#[cfg(feature = "optimism")]
+pub mod optimism;
+
+use crate::{ForkCondition, Hardfork};
+
+/// Dev hardforks
+pub static DEV_HARDFORKS: [(Hardfork, ForkCondition); 14] = [
```
Comment (mattsse) [inline crates/ethereum-forks/src/chains/mod.rs:11]: this should be under dev.rs

File: crates/ethereum-forks/src/chains/optimism.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+use crate::{ForkCondition, Hardfork};
+use alloy_primitives::U256;
+
+/// Optimism mainnet hardforks
+pub static OP_MAINNET_HARDFORKS: [(Hardfork, ForkCondition); 21] = [
```
Comment (mattsse) [inline crates/ethereum-forks/src/chains/optimism.rs:5]: we probably don't need static?
wdyt @DaniPopes

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic nits.

supportive of name change as followup

File: crates/ethereum-forks/src/chains/ethereum.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
+use crate::{ForkCondition, Hardfork};
+use alloy_primitives::U256;
+
+const PARIS_TTD_MAINNET: U256 = U256::from_limbs([15566869308787654656, 3184, 0, 0]);
+const PARIS_TTD_GOERLI: U256 = U256::from_limbs([10790000, 0, 0, 0]);
+const PARIS_TTD_SEPOLIA: U256 = U256::from_limbs([17000000000000000, 0, 0, 0]);
```
Comment (DaniPopes) [inline crates/ethereum-forks/src/chains/ethereum.rs:6]: use `uint!` macro

File: crates/ethereum-forks/src/chains/ethereum.rs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        let ttd = U256::from(U_PARIS_TTD_MAINNET);
+        assert_eq!(ttd, PARIS_TTD_MAINNET);
+
+        // goerli
+        let ttd = U256::from(U_PARIS_TTD_GOERLI);
+        assert_eq!(ttd, PARIS_TTD_GOERLI);
+
+        // sepolia
+        let ttd = U256::from(U_PARIS_TTD_SEPOLIA);
+        assert_eq!(ttd, PARIS_TTD_SEPOLIA);
```
Comment (DaniPopes) [inline crates/ethereum-forks/src/chains/ethereum.rs:109]: no need with `uint!` macro

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm

pending uint! and @DaniPopes

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**